components/clarity-lsp/src/vscode_bridge.rs

```diff
@@ -164,14 +165,15 @@ impl LspVscodeBridge {
 
     #[wasm_bindgen(js_name=onRequest)]
     pub fn request_handler(&self, method: String, js_params: JsValue) -> Result<JsValue, JsValue> {
+        let serializer = Serializer::json_compatible();
         match method.as_str() {
             Initialize::METHOD => {
                 let lsp_response = process_request(
                     LspRequest::Initialize(decode_from_js(js_params)?),
                     &EditorStateInput::RwLock(self.editor_state_lock.clone()),
                 );
                 if let LspRequestResponse::Initialize(response) = lsp_response {
-                    return encode_to_js(&response).map_err(|_| JsValue::NULL);
+                    return response.serialize(&serializer).map_err(|_| JsValue::NULL);
```

Comment:
The regular `to_value()` (`encode_to_js`)  methods converts Rust structs to native JS maps.
But in this case, the lsp client expects JS literal objects, which is what  `let serializer = Serializer::json_compatible();` does